File: androiddagger2example/app/src/main/java/org/sinabro/android_dagger2_example/module/AppModule.java
package org.sinabro.android_dagger2_example.module;

import android.app.Application;

import javax.inject.Singleton;

import dagger.Provides;

/**
 * Created by 윤정현 on 2018-02-06.
 */

public class AppModule {
    private Application application;

    AppModule(Application application){
        this.application=application;
    }

    @Provides
    @Singleton
    Application provideApplication(){
        return this.application;
    }
}

File: app/src/main/java/android/dagger2/demo/android_kotlin_dagger2/module/AppModule.kt
package android.dagger2.demo.android_kotlin_dagger2.module

import android.app.Application
import dagger.Module
import dagger.Provides
import javax.inject.Singleton


@Module
class AppModule(private val app: Application) {

    @Provides
    @Singleton
    fun provideApp() = app
}